core/templates/pages/story-editor-page/editor-tab/story-node-editor.component.ts

```diff
@@ -203,13 +221,51 @@ export class StoryNodeEditorComponent implements OnInit, OnDestroy {
       this.storyUpdateService.setStoryNodeDescription(
         this.story, this.nodeId, newDescription);
       this.currentDescription = newDescription;
+      this.updateCurrentNodeIsPublishable();
     }
   }
 
+  updatePlannedPublicationDate(dateString: string | null): void {
+    let newPlannedPublicationDate = dateString && dateString.length ? new Date(
```

Comment:
btw, `empty_string && anything` evaluates to false. So maybe `dateString ? .. : ..` may be sufficient.

Also, I'm wondering, in what scenario would dateString be empty? If it isn't a valid input, perhaps we should consider not processing further (i.e. add an explicit if condition and return)?

Reply:
Updated.
And also dateString will not be an empty string, so I have kept what you have suggested in the 1st line.